content/docs/thinking-in-react.md

```diff
@@ -31,118 +31,118 @@ Our JSON API returns some data that looks like this:
+你可以自上而下或者自下而上构建应用：自上而下意味着首先编写层次较高的组件（比如示例中的 `FilterableProductTable`），自下而上意味着从最基本的组件开始编写（比如示例中的 `ProductRow`）。当你的应用比较简单时，使用自上而下的方式更方便；对于较为庞大的项目来说，自下而上地构建，并同时为低层组件编写测试是比较合适的方式。
 
-At the end of this step, you'll have a library of reusable components that render your data model. The components will only have `render()` methods since this is a static version of your app. The component at the top of the hierarchy (`FilterableProductTable`) will take your data model as a prop. If you make a change to your underlying data model and call `ReactDOM.render()` again, the UI will be updated. It's easy to see how your UI is updated and where to make changes since there's nothing complicated going on. React's **one-way data flow** (also called *one-way binding*) keeps everything modular and fast.
+到此为止，你应该已经有了一个可重用的组件库来渲染你的数据模型。由于我们构建的是静态版本，这些组件应该只有唯一的一个 `render()` 方法。最顶层的组件 `FilterableProductTable` 通过 props 接受你的数据模型。一旦你的数据模型发生了改变，`ReactDOM.render()` 就会被调用，从而更新 UI。所幸这个组件并不复杂，所以很容易修改，也很容易看到 UI 的具体改变。React **单向数据流**（也叫*单向绑定*）的思想使得组件模块化，易于快速开发。
 
-Simply refer to the [React docs](/docs/) if you need help executing this step.
+如果你在完成这一步骤时遇到了困难，可以参阅 [React 文档](/docs/)。
 
-### A Brief Interlude: Props vs State
+### 补充说明: 有关 props 和 state
 
-There are two types of "model" data in React: props and state. It's important to understand the distinction between the two; skim [the official React docs](/docs/interactivity-and-dynamic-uis.html) if you aren't sure what the difference is.
+在 React 中，有两种数据“模型”：props 和 state。清楚地理解两者的区别是十分重要的；如果你不太有把握，可以参阅 [React 官方文档](/docs/interactivity-and-dynamic-uis.html)。
 
-## Step 3: Identify The Minimal (but complete) Representation Of UI State
+## 第三步：确保 UI state 为最小（且完整）集合
 
-To make your UI interactive, you need to be able to trigger changes to your underlying data model. React makes this easy with **state**.
+为了使你的 UI 具有交互功能，你需要及时地触发底层数据模型的改变。React 通过 **state** 来完成这个任务。
 
-To build your app correctly, you first need to think of the minimal set of mutable state that your app needs. The key here is [DRY: *Don't Repeat Yourself*](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). Figure out the absolute minimal representation of the state your application needs and compute everything else you need on-demand. For example, if you're building a TODO list, just keep an array of the TODO items around; don't keep a separate state variable for the count. Instead, when you want to render the TODO count, simply take the length of the TODO items array.
+为了正确地构建应用，你首先需要考虑应用所需的最小可变 state 集合。其中的关键正是 [DRY: *Don't Repeat Yourself*](https://zh.wikipedia.org/wiki/%E4%B8%80%E6%AC%A1%E4%B8%94%E4%BB%85%E4%B8%80%E6%AC%A1)。只保留应用需要的最小 state 集合，其他数据均由它们计算产生。比如，你要编写一个任务清单应用，你只需要保存一个包含所有事项的数组，而不需要额外保存数组的长度（任务个数）。当你需要展示任务个数时，只需要利用该数组的 length 属性即可。
```

Comment:
What did you do in the old docs? I think it's up to the translator: if you think the Chinese version is still informative enough, keep it. Otherwise link to the English docs.

Reply:
Well, then let's keep it the same as before (not changed) when the link directs to a Wikipedia site. We should claim it somewhere? @QC-L